.github/workflows/integration-tests-kubernetes-core.yaml

```diff
@@ -0,0 +1,143 @@
+---
+name: Integration tests
+on:
+  pull_request:
+    branches:
+      - main
+      - stable-*
+
+jobs:
+  splitter:
+    env:
+      source_dir: "./source"
+    runs-on: ubuntu-latest
+    outputs:
+      test_targets: ${{ steps.splitter.outputs.test_targets }}
+      test_targets_json: ${{ steps.splitter.outputs.test_targets_json }}
+      test_jobs: ${{ steps.splitter.outputs.test_jobs }}
+    steps:
+      - name: Checkout the collection repository
+        uses: actions/checkout@v3
+        with:
+          repository: ansible-collections/kubernetes.core
+          ref: main
+          path: ${{ env.source_dir }}
+          fetch-depth: "0"
+
+      - name: Split all Kubernetes targets into Job
+        id: splitter
+        uses: ansible-network/github_actions/.github/actions/ansible_test_splitter@main
+        with:
+          collections_to_test: ${{ env.source_dir }}
+          total_jobs: 8
+        env:
+          ANSIBLE_TEST_ALL_THE_TARGETS: "true"
+
+      - name: Display splitter output
+        run: |
+          echo "test_targets=${{ steps.splitter.outputs.test_targets }}"
+          echo "test_targets_json=${{ steps.splitter.outputs.test_targets_json }}"
+          echo "test_jobs=${{ steps.splitter.outputs.test_jobs }}"
+        shell: bash
+  integration:
+      runs-on: ubuntu-latest
+      timeout-minutes: 60
+      needs:
+        - splitter
+      if: ${{ needs.splitter.outputs.test_targets != '' }}
+      env:
+        ansible_version: "milestone"
+        python_version: "3.11"
```

Comment:
Should this match the matrix python version?

Reply:
this one is not used, I will remove it

Reply:
updated